payment/src/main/java/org/killbill/billing/payment/core/sm/control/ControlPluginRunner.java

```diff
@@ -117,7 +119,7 @@ public PriorPaymentControlResult executePluginPriorCalls(final Account account,
                 inputPluginProperties = prevResult.getAdjustedPluginProperties();
             }
             if (prevResult.isAborted()) {
-                break;
+                throw new PaymentApiException(ErrorCode.PAYMENT_PLUGIN_API_ABORTED, pluginName);
             }
```

Comment:
Too bad we don't have the reason for the abort. I agree passing the pluginName is good, but maybe format a better message (e.g: "plugin name= 'foo'").